monitor/api/src/main/java/resources/Monitor.java

```diff
@@ -0,0 +1,874 @@
+	}
+
+
+	@POST
+	@Produces("text/plain")
+	@Path("/addObserver/{metricName}/{portOfObserver}")//http://localhost:8080/monitor-api/rest/addObserver/metricName/portOfObserver callbackURL
+	public String addObserver( @PathParam("metricName") String metricName, @PathParam("portOfObserver") String portOfObserver, String callbackURL ) {
+
+		//System.out.println( "metricName= " + metricName );
+		//System.out.println( "portOfObserver= " + portOfObserver );
+		//System.out.println( "callbackURL= " + callbackURL );
+
+
+		Controller controller = new ControllerImpl();
+
+		String msg = controller.addObserver( metricName, portOfObserver, callbackURL );
+
+
+		return "[INFO] Monitor REST Service: Adding observer...\n\n" + msg;
+	}
+
+
+	@POST
+	@Produces("text/plain")
+	@Path("/sendReplanningEvent")//http://localhost:8080/monitor-api/rest/sendReplanningEvent xxxxxx
+	public String sendReplanningEvent( String replanningEvent ) {
+
+		//System.out.println( "\nreplanningEvent = " + replanningEvent );
+
+
+		String msg = notifyForReplanningEvent( replanningEvent );
+
+		if( msg == null ) msg = "";
+
+
+		return "[INFO] Monitor REST Service: Sending replanning event...\n\n" + replanningEvent + "\n" + msg;
+	}
+
+	private String notifyForReplanningEvent( String replanningEvent ) {
+
+		String msg = "", msg1 = "", msg2 = "", msg3 = "";
+
+
+		Properties properties = read( INITIALIZATION_CONFIGURATION_FILE_ON_SERVER );
+
+		if( properties == null ) msg = "[ERROR] Monitor REST Service: there is no calling information about the external SeaClouds components.";
+
+		else{
+
+			try{
+
+				String SLAServiceURIReplanning = properties.getProperty( "SLAServiceURIReplanning" ); //System.out.println( "\nSLAServiceURIRulesReady = " + SLAServiceURIRulesReady );
+				String DashboardURIReplanning = properties.getProperty( "DashboardURIReplanning" );
+				String PlannerURIReplanning = properties.getProperty( "PlannerURIReplanning" );
+
+				if( SLAServiceURIReplanning == null ) msg1 = "[ERROR] Monitor REST Service: there is no calling information about the SLA Service.";
+
+				else 	msg1 = RESTPost.httpPost( SLAServiceURIReplanning, replanningEvent, "xml" );
+
+				if( msg1 != null ) msg += msg1; //System.out.println( "\nmsg1 = " + msg1 ); System.out.println( "\nmsg = " + msg );
+
+
+				if( DashboardURIReplanning == null ) msg2 = "[ERROR] Monitor REST Service: there is no calling information about the Dashboard.";
+
+				else 	msg2 = RESTPost.httpPost( DashboardURIReplanning, replanningEvent, "xml" );
+
+				if( msg2 != null ) msg += "\n" + msg2;
+
+
+				if( PlannerURIReplanning == null ) msg3 = "[ERROR] Monitor REST Service: there is no calling information about the Planner.";
+
+				else 	msg3 = RESTPost.httpPost( PlannerURIReplanning, replanningEvent, "xml" );
+
+				if( msg3 != null ) msg += "\n" + msg3;
+			}
+
+			catch( Exception ex ){
+
+				ex.printStackTrace();
+			}
+		}
+
+
+		return msg;
+	}
+
+
+	public static void main( String[] args ) {
```

Comment:
This main method looks like it serves a different purpose than the rest of this source file. Would it make sense to keep it in a different file, something like MonitorCLI?